test/integration/controller/jobs/deployment/deployment_webhook_test.go

```diff
@@ -0,0 +1,131 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/discovery"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	configapi "sigs.k8s.io/kueue/apis/config/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/constants"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/util/kubeversion"
+	testingdeployment "sigs.k8s.io/kueue/pkg/util/testingjobs/deployment"
+	"sigs.k8s.io/kueue/test/integration/framework"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+var _ = ginkgo.Describe("Deployment Webhook", func() {
+	var ns *corev1.Namespace
+
+	ginkgo.When("with pod integration enabled", ginkgo.Ordered, ginkgo.ContinueOnFailure, func() {
+		ginkgo.BeforeAll(func() {
+			fwk = &framework.Framework{
+				CRDPath:     crdPath,
+				WebhookPath: webhookPath,
+			}
+			cfg = fwk.Init()
+
+			discoveryClient, err := discovery.NewDiscoveryClientForConfig(cfg)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			serverVersionFetcher = kubeversion.NewServerVersionFetcher(discoveryClient)
+			err = serverVersionFetcher.FetchServerVersion()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ctx, k8sClient = fwk.RunManager(cfg, managerSetup(
+				nil,
+				jobframework.WithManageJobsWithoutQueueName(false),
+				jobframework.WithKubeServerVersion(serverVersionFetcher),
+				jobframework.WithIntegrationOptions(corev1.SchemeGroupVersion.WithKind("Pod").String(), &configapi.PodIntegrationOptions{
+					PodSelector: &metav1.LabelSelector{},
+					NamespaceSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      "kubernetes.io/metadata.name",
+								Operator: metav1.LabelSelectorOpNotIn,
+								Values:   []string{"kube-system", "kueue-system"},
+							},
+						},
+					},
+				}),
+			))
+		})
+		ginkgo.BeforeEach(func() {
+			ns = &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "deployment-",
+				},
+			}
+			gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		})
+		ginkgo.AfterAll(func() {
+			fwk.Teardown()
+		})
+
+		ginkgo.When("The queue-name label is set", func() {
+			var (
+				deployment *appsv1.Deployment
+				lookupKey  types.NamespacedName
+			)
+
+			ginkgo.BeforeEach(func() {
+				deployment = testingdeployment.MakeDeployment("deployment-with-queue-name", ns.Name).
+					Queue("user-queue").
+					Obj()
+				lookupKey = types.NamespacedName{Name: deployment.Name, Namespace: deployment.Namespace}
+			})
+
+			ginkgo.It("Should inject queue name to pod template labels", func() {
+				gomega.Expect(k8sClient.Create(ctx, deployment)).Should(gomega.Succeed())
+
+				createdDeployment := &appsv1.Deployment{}
+				gomega.Eventually(func() error {
+					return k8sClient.Get(ctx, lookupKey, createdDeployment)
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
```

Comment:
```suggestion
				gomega.Eventually(func(g gomega.Gomega) error {
					createdDeployment := &appsv1.Deployment{}
					g.Expect(k8sClient.Get(ctx, lookupKey, createdDeployment)).Should(gomega.Succeeded())
					g.Expect(createdDeployment.Spec.Template.Labels[constants.QueueLabel]).To(gomega.Equal("user-queue"),
					"Queue name should be injected to pod template labels")
				}, util.Timeout, util.Interval).Should(gomega.Succeed())
```

We can avoid exposing the `createdDeployment`.